buildrunner/validation/config.py

```diff
@@ -132,6 +138,155 @@ def validate_push(
+                            images.append(f"{pushcommmit.repository}:{tag}")
+                    else:
+                        images.append(f"{pushcommmit.repository}:latest")
+                elif isinstance(pushcommmit, str):
+                    if ":" not in pushcommmit:
+                        images.append(f"{pushcommmit}:latest")
+                    else:
+                        images.append(pushcommmit)
+                elif isinstance(pushcommmit, list):
+                    for item in pushcommmit:
+                        if isinstance(item, str):
+                            images.append(item)
+                        elif isinstance(item, StepPushCommitDict):
+                            if item.tags:
+                                for tag in item.tags:
+                                    images.append(f"{item.repository}:{tag}")
+                            else:
+                                images.append(f"{item.repository}:latest")
+                        else:
+                            raise ValueError(
+                                f"Unknown type for step.push: {type(step.push)}"
+                            )
+                return images
+
+            images = []
+            images.extend(get_images(step.push))
+            images.extend(get_images(step.commit))
+            return images
+
+        def validate_multiplatform_are_not_retagged():
+            """
+            Validate multi-platform are not re-tagged
+
+            Args:
+                mp_push_tags (Set[str]): Set of all tags used in multi-platform build steps
+
+            Raises:
+                ValueError | pydantic.ValidationError: If the config file is invalid
+            """
+            step_images = {}
+
+            # Iterate through each step and get information about the images
+            for step_name, step in vals.items():
+                src_image, is_multi_platform = get_source_image(step)
+                dst_images = get_destination_images(step)
+                step_images[step_name] = StepImagesInfo(
+                    source_image=src_image,
+                    dest_images=dst_images,
+                    is_multi_platform=is_multi_platform,
+                )
+
+            # Iterate through each step images and check for multi-platform re-tagging
+            retagged_images = []
+            for step_name, step_images_info in step_images.items():
+                other_steps_images_infos = [
+                    curr_step_images_info
+                    for curr_step_name, curr_step_images_info in step_images.items()
+                    if curr_step_name != step_name
+                ]
+                for other_step_info in other_steps_images_infos:
+                    for src_image in step_images_info.source_image:
+                        if (
+                            src_image in other_step_info.dest_images
+                            and step_images_info.dest_images
```

Comment:
I feel like this check should be at the top of the for loop:
```
if not step_images_info.dest_images:
    continue
```

Reply:
I think it's logically equivalent. But I do think your suggestion to break it up makes it more readable and efficient.  So I made that change.